backend/handlers/bets/betutils/feeutils_test.go

```diff
@@ -0,0 +1,198 @@
+	}
+	db.Create(&bets) // Save the bet to the database
+
+	// Scenario 2: User places another bet on Market 1 where they already have a bet
+	initialBetFee = getUserInitialBetFee(db, 1, user)
+	if initialBetFee != 0 {
+		t.Errorf("Expected initial bet fee to be 0, got %d", initialBetFee)
+	}
+
+	// Scenario 3: User places a bet on Market 2 where they have no prior bets
+	initialBetFee = getUserInitialBetFee(db, 2, user)
+	if initialBetFee != appConfig.Economics.Betting.BetFees.InitialBetFee {
+		t.Errorf("Expected initial bet fee to be %d, got %d", appConfig.Economics.Betting.BetFees.InitialBetFee, initialBetFee)
+	}
+
+}
+
+func TestGetTransactionFee(t *testing.T) {
+	// Mock the appConfig with test data
+	appConfig = mockEconomicConfig()
+
+	// Test buy scenario
+	buyBet := models.Bet{Amount: 100}
+	transactionFee := getTransactionFee(buyBet)
+	if transactionFee != appConfig.Economics.Betting.BetFees.BuySharesFee {
+		t.Errorf("Expected buy transaction fee to be %d, got %d", appConfig.Economics.Betting.BetFees.BuySharesFee, transactionFee)
+	}
+
+	// Test sell scenario
+	sellBet := models.Bet{Amount: -100}
+	transactionFee = getTransactionFee(sellBet)
+	if transactionFee != appConfig.Economics.Betting.BetFees.SellSharesFee {
+		t.Errorf("Expected sell transaction fee to be %d, got %d", appConfig.Economics.Betting.BetFees.SellSharesFee, transactionFee)
+	}
+}
+
+func TestGetSumBetFees(t *testing.T) {
+	// Set up in-memory SQLite database
+	db, err := gorm.Open(sqlite.Open(":memory:"), &gorm.Config{})
+	if err != nil {
+		t.Fatalf("Failed to connect to the database: %v", err)
+	}
+
+	// Migrate the Bet model
+	db.AutoMigrate(&models.Bet{})
+
+	// Mock the appConfig with test data
+	appConfig = mockEconomicConfig()
+
+	// Create a test user
+	user := &models.User{Username: "testuser"}
+
+	// Scenario 1: User has no bets, buys shares, gets initial fee
+	buyBet := models.Bet{MarketID: 1, Amount: 100}
+	sumOfBetFees := GetSumBetFees(db, user, buyBet)
+	expectedSum := appConfig.Economics.Betting.BetFees.InitialBetFee
+	if sumOfBetFees != expectedSum {
+		t.Errorf("Expected sum of bet fees to be %d, got %d", expectedSum, sumOfBetFees)
+	}
+
+	// Create a test bet
+	bets := []models.Bet{
+		{Username: "testuser", MarketID: 1, Amount: 100, PlacedAt: time.Now()},
+	}
+	db.Create(&bets)
+
+	// Scenario 2: User has one bet, buys shares
+	sumOfBetFees = GetSumBetFees(db, user, buyBet)
+	expectedSum = appConfig.Economics.Betting.BetFees.BuySharesFee
+	if sumOfBetFees != expectedSum {
+		t.Errorf("Expected sum of bet fees to be %d, got %d", expectedSum, sumOfBetFees)
+	}
+
+	// Scenario 3: User has one bet, sells shares
+	sellBet := models.Bet{MarketID: 1, Amount: -100}
+	sumOfBetFees = GetSumBetFees(db, user, sellBet)
+	expectedSum = appConfig.Economics.Betting.BetFees.SellSharesFee
+	if sumOfBetFees != expectedSum {
+		t.Errorf("Expected sum of bet fees to be %d, got %d", expectedSum, sumOfBetFees)
+	}
```

Comment:
I'm honestly still struggling to understand the selling shares. Should a user be able to sell more than they've bought?

Reply:
No, user should only be able to sell their position which is calculated in https://github.com/openpredictionmarkets/socialpredict/blob/main/backend/handlers/positions/positionsmath.go

I changed this so that the user is only selling one share.

Admittedly I believe there are some marginal buying and selling problems which lead to market probabilities that don't make sense (e.g. buy and sell as a single trader an amount that leaves you at 0 shares, yet the probability may be +/- 0.5, which, it should be back to 0.5).

I will have to create some kind of diagram that helps explain positions calculation and dbpm. I have this written in a notebook from a while ago somewhere so maybe I can just take some snapshots of that to start off with.